### PR TITLE
Add asynchronous metrics on MergeTree tables data volume

### LINKED ISSUE #11714: Asynchronous metrics on data volume
We can add metrics on total amount of rows and bytes in MergeTree tables.
It involves lock of data parts and iteration over all of them. We can make it more efficient by maintaining additional variable with that value.

### PR DESCRIPTION
**Pull request by ucasfl**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add asynchronous metrics on total amount of rows, bytes and parts in MergeTree tables.
This fix [#11714](https://github.com/ClickHouse/ClickHouse/issues/11714)

Detailed description / Documentation draft:

Add metrics on total MergeTree Tables data volume into AsynchronousMetrics.

### CODE REVIEW

**Comment by ucasFL** [comment]
I don't know how to test this. Because AsynchronousMetrics updated each minute, it's too long, and system tables are dynamically changing?

**Review by Akazz — COMMENTED** [review]
This PR looks okay in the rest of it. Although I am a bit uneasy on changing `MergeTreeData::getTotalActiveSizeInBytes()` and two others b/c they might lose their guarantees of returning consistent values. Let us see how it plays out.

File: src/Interpreters/AsynchronousMetrics.cpp [hunk 1]
```diff
@@ -293,6 +297,10 @@ void AsynchronousMetrics::update()
                 if (table_merge_tree)
                 {
                     calculateMax(max_part_count_for_partition, table_merge_tree->getMaxPartsCountForPartition());
+                    const auto settings = context.getSettingsRef();
+                    total_number_of_bytes += table_merge_tree->totalBytes(settings).value();
+                    total_number_of_rows += table_merge_tree->totalRows(settings).value();
+                    total_number_of_parts += table_merge_tree->getPartsCount();
```
[the remark this answers is not in the document]
Comment (ucasFL) [reply src/Interpreters/AsynchronousMetrics.cpp:303]: Ok, i will fix it.

**Comment by Akazz** [comment]
There is something wrong with `test_ttl_replicated::test_ttl_empty_parts`. And that is due to changes in `MergeTreeData::getPartsCount()`.

**Comment by ucasFL** [comment]
> There is something wrong with `test_ttl_replicated::test_ttl_empty_parts`. And that is due to changes in `MergeTreeData::getPartsCount()`.

I think problem is happend in here, when different replica simultaneously drop an empty parts.

https://github.com/ClickHouse/ClickHouse/blob/5e860200b1fede2cb43a09134c6de2482ab30ff0/src/Storages/MergeTree/MergeTreeData.cpp#L1225-L1243

**Comment by ucasFL** [comment]
Looks ok now.

**Review by Akazz — APPROVED** [review]
LGTM

But we should keep an eye on the behavior of  `count()` since it uses `MergeTreeData::getTotalActiveSizeInRows()` under the hood, and the method has changed its consistency guarantees with this PR.

**Comment by Akazz** [comment]
All of the failed performance tests are due to "Unexpected Query Duration" and that's irrelevant to this PR

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 2]
```diff
@@ -2037,7 +2058,11 @@ void MergeTreeData::removePartsFromWorkingSet(const MergeTreeData::DataPartsVect
     for (const DataPartPtr & part : remove)
     {
         if (part->state == IMergeTreeDataPart::State::Committed)
+        {
             removePartContributionToColumnSizes(part);
+            /// Update data volume
```
[the remark this answers is not in the document]
Comment (ucasFL) [reply src/Storages/MergeTree/MergeTreeData.cpp:2063]: Ok, I will clear these redundant comment.

**Pull request merged**